# 20221230

```bash
 crio.service - Container Runtime Interface for OCI (CRI-O)
     Loaded: loaded (/lib/systemd/system/crio.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2022-12-30 19:25:46 JST; 27s ago
       Docs: https://github.com/cri-o/cri-o
   Main PID: 4665 (crio)
      Tasks: 12
     Memory: 11.1M
        CPU: 443ms
     CGroup: /system.slice/crio.service
             └─4665 /usr/bin/crio

Dec 30 19:25:46 k8s-master crio[4665]: time="2022-12-30 19:25:46.313191648+09:00" level=info msg="Using seccomp default profile when unspecified: true"
Dec 30 19:25:46 k8s-master crio[4665]: time="2022-12-30 19:25:46.313284295+09:00" level=info msg="No seccomp profile specified, using the internal default"
Dec 30 19:25:46 k8s-master crio[4665]: time="2022-12-30 19:25:46.313327813+09:00" level=info msg="Installing default AppArmor profile: crio-default"
Dec 30 19:25:46 k8s-master crio[4665]: time="2022-12-30 19:25:46.417856344+09:00" level=info msg="No blockio config file specified, blockio not configured"
Dec 30 19:25:46 k8s-master crio[4665]: time="2022-12-30 19:25:46.418095397+09:00" level=info msg="RDT not available in the host system"
Dec 30 19:25:46 k8s-master crio[4665]: time="2022-12-30 19:25:46.419546029+09:00" level=warning msg="Error validating CNI config file /etc/cni/net.d/100-crio-bridge.conf: [failed to find plugin \"bridge\" in path [/opt/cni/bin/]]"
Dec 30 19:25:46 k8s-master crio[4665]: time="2022-12-30 19:25:46.419823582+09:00" level=warning msg="Error validating CNI config file /etc/cni/net.d/200-loopback.conf: [failed to find plugin \"loopback\" in path [/opt/cni/bin/]]"
Dec 30 19:25:46 k8s-master crio[4665]: time="2022-12-30 19:25:46.419880563+09:00" level=info msg="Updated default CNI network name to "
Dec 30 19:25:46 k8s-master crio[4665]: time="2022-12-30 19:25:46.489792155+09:00" level=warning msg="Error encountered when checking whether cri-o should wipe images: version file /var/lib/crio/version not found: open /var/lib/crio/version: no such file or directory"
Dec 30 19:25:46 k8s-master systemd[1]: Started Container Runtime Interface for OCI (CRI-O).
~
```

```bash
ubuntu@k8s-master:~$ cat /boot/firmware/cmdline.txt 
console=serial0,115200 dwc_otg.lpm_enable=0 console=tty1 root=LABEL=writable rootfstype=ext4 rootwait fixrtc quiet splash cgroup_enable=cpuset cgroup_memory=1 cgroup_enable=memory
```

```bash
ubuntu@k8s-master:~$ kubectl logs -n kube-flannel kube-flannel-ds-bsk5d
Defaulted container "kube-flannel" out of: kube-flannel, install-cni-plugin (init), install-cni (init)
I1230 16:32:19.530189       1 main.go:204] CLI flags config: {etcdEndpoints:http://127.0.0.1:4001,http://127.0.0.1:2379 etcdPrefix:/coreos.com/network etcdKeyfile: etcdCertfile: etcdCAFile: etcdUsername: etcdPassword: version:false kubeSubnetMgr:true kubeApiUrl: kubeAnnotationPrefix:flannel.alpha.coreos.com kubeConfigFile: iface:[] ifaceRegex:[] ipMasq:true ifaceCanReach: subnetFile:/run/flannel/subnet.env publicIP: publicIPv6: subnetLeaseRenewMargin:60 healthzIP:0.0.0.0 healthzPort:0 iptablesResyncSeconds:5 iptablesForwardRules:true netConfPath:/etc/kube-flannel/net-conf.json setNodeNetworkUnavailable:true}
W1230 16:32:19.530464       1 client_config.go:617] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I1230 16:32:19.563935       1 kube.go:126] Waiting 10m0s for node controller to sync
I1230 16:32:19.564027       1 kube.go:431] Starting kube subnet manager
I1230 16:32:20.564372       1 kube.go:133] Node controller sync successful
I1230 16:32:20.564484       1 main.go:224] Created subnet manager: Kubernetes Subnet Manager - k8s-master
I1230 16:32:20.564525       1 main.go:227] Installing signal handlers
I1230 16:32:20.565432       1 main.go:467] Found network config - Backend type: vxlan
I1230 16:32:20.565876       1 match.go:206] Determining IP address of default interface
I1230 16:32:20.568304       1 match.go:259] Using interface with name eth0 and address 192.168.10.101
I1230 16:32:20.568494       1 match.go:281] Defaulting external address to interface address (192.168.10.101)
I1230 16:32:20.568753       1 vxlan.go:138] VXLAN config: VNI=1 Port=0 GBP=false Learning=false DirectRouting=false
E1230 16:32:20.578821       1 main.go:327] Error registering network: operation not supported
I1230 16:32:20.578987       1 main.go:447] Stopping shutdownHandler...
```

kubectl apply -f https://raw.githubusercontent.com/flannel-io/flannel/v0.20.2/Documentation/kube-flannel.yml

- [Fail running on Raspberry Pi Ubuntu 21.10](https://github.com/k3s-io/k3s/issues/4234)

```bash
sudo apt install linux-modules-extra-raspi && reboot
```

## sed

```bash
ubuntu@k8s-worker2:~$ cat /boot/firmware/cmdline.txt 
console=serial0,115200 dwc_otg.lpm_enable=0 console=tty1 root=LABEL=writable rootfstype=ext4 rootwait fixrtc quiet splash
```

```bash
sed -i "s/splash$/splash cgroup_enable=cpuset cgroup_memory=1 cgroup_enable=memory/" /boot/firmware/cmdline.txt
```

## kubeadm complains about bridge-nf-call and ip_forward if not using docker runtime

- [kubeadm complains about bridge-nf-call and ip_forward if not using docker runtime](https://github.com/kubernetes/kubeadm/issues/1062)

## error execution phase preflight: couldn't validate the identity of the API Server: Get "https://192.168.1.101:6443/api/v1/namespaces/kube-public/configmaps/cluster-info?timeout=10s": net/http: request canceled while waiting for connection (Client.

```bash
ubuntu@k8s-worker1:~$ sudo kubeadm join 192.168.1.101:6443 --token 9y44r8.apdue7u8vzd0wpfz --discovery-token-ca-cert-hash sha256:3e9ef8910b95e0a366041c1e156b7cbd6802df4c857cd53ad59bbba631749983
[preflight] Running pre-flight checks
error execution phase preflight: [preflight] Some fatal errors occurred:
        [ERROR FileContent--proc-sys-net-bridge-bridge-nf-call-iptables]: /proc/sys/net/bridge/bridge-nf-call-iptables does not exist
[preflight] If you know what you are doing, you can make a check non-fatal with `--ignore-preflight-errors=...`
To see the stack trace of this error execute with --v=5 or higher
ubuntu@k8s-worker1:~$ modprobe br_netfilter
modprobe: ERROR: could not insert 'br_netfilter': Operation not permitted
ubuntu@k8s-worker1:~$ sudo modprobe br_netfilter
ubuntu@k8s-worker1:~$ sudo kubeadm join 192.168.1.101:6443 --token 9y44r8.apdue7u8vzd0wpfz --discovery-token-ca-cert-hash sha256:3e9ef8910b95e0a366041c1e156b7cbd6802df4c857cd53ad59bbba631749983
[preflight] Running pre-flight checks
error execution phase preflight: couldn't validate the identity of the API Server: Get "https://192.168.1.101:6443/api/v1/namespaces/kube-public/configmaps/cluster-info?timeout=10s": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
To see the stack trace of this error execute with --v=5 or higher
```

- [クラスターに参加するノードとコントロールプレーンノード間で相互信頼を確立するためのkubeadmトークンの有効期限が切れてしまっていることが原因](https://qiita.com/yasubehe/items/e4f212633723f0d1837d)

```bash
ubuntu@k8s-master:~$ kubeadm token create --print-join-command
kubeadm join 192.168.10.101:6443 --token 3u2z7v.qx81p3azvu15ftzw --discovery-token-ca-cert-hash sha256:0e731a79605ba03cfbc823e86e8b81b2fcdcf7f1887c1d2d86239ed87fff8d04 
```

新しくしてみた。

```bash
ubuntu@k8s-worker1:~$ sudo kubeadm join 192.168.1.101:6443 --token 3u2z7v.qx81p3azvu15ftzw --discovery-token-ca-cert-hash sha256:0e731a79605ba03cfbc823e86e8b81b2fcdcf7f1887c1d2d86239ed87fff8d04
[preflight] Running pre-flight checks
error execution phase preflight: couldn't validate the identity of the API Server: Get "https://192.168.1.101:6443/api/v1/namespaces/kube-public/configmaps/cluster-info?timeout=10s": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
To see the stack trace of this error execute with --v=5 or higher
```

--v=5 をつけてやったらできた。謎
